MetadataExtractor/IO/BufferReader.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Buffers.Binary;
+
+namespace MetadataExtractor.IO;
+
+internal ref struct BufferReader(ReadOnlySpan<byte> bytes, bool isBigEndian)
```

Comment:
What do you think about calling this `SequentialReader`, for consistency with the other reader classes. We might one day have an `IndexedBufferReader` too.

Reply:
I had given some thought on that as well.

My original plan was to extend the BufferReader with indexed methods, rather than duplicating.

For example:

```csharp
ref struct BufferReader
{
  GetBytes(span) // auto-advanced sequential access
  GetBytes(index, span) // direct access
  GetBytes(reader.Position + 4, span); // look ahead without advancing
}
```

Let me know what you think about combining these access methods in a single reader, or if you think we should make a hard distinction. By combining, we also allow consumers to mix the two patterns (e.g. looking ahead). 

In non-continuous memory, the distinction seems more critical to constrain the consumers (and avoid unintended stream advancement, buffering, tearing, etc).

Reply:
That makes sense for this type. It can comfortably support both read patterns.